rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
 		if (selection.size > 1) {
 			return selection.toList.map[testCaseListFromSelection].flatten.toList
 		} else {
-			if (selection.firstElement instanceof IFolder) {
-				return selection.firstElement.testCaseListFromSelection.toList
-			}
-			return #[launchShortcutUtil.getQualifiedNameForTestInTcl(selection.firstElement as IResource).toString]
+			return selection.firstElement.testCaseListFromSelection.toList
```

Comment:
The if statement checks the selection size. The logic was dublicated in the further version the function handles the multiple selection.